internal/command/jobs/printers.go

```diff
@@ -119,6 +125,79 @@ func printJobDetails(cfg *config.Job, out io.Writer, job *models.Job) error {
 	return nil
 }
 
+func waitForJob(ctx context.Context, cfg *config.JobSubmit, out io.Writer, jobName string) error {
+
+	fmt.Fprintf(out, "Waiting for job execution\n")
+
+	looped := false
+
+	retry := poll.
+		NewPoll().
+		WithDelay(TimeBetweenJobRefresh).
+		WithTimeout(cfg.Timeout)
+
+	lastCursor := ""
+
+	err := retry.Until(func() bool {
+		j, err := getJob(cfg.Job, jobName)
+		if err != nil {
+			fmt.Fprintf(out, "Error getting job: %s", err.Error())
+
+			// We want to keep retrying if the timeout has not been exceeded
+			return false
+		}
+
+		attempt := j.Status.Attempts[0]
+		switch attempt.Phase {
+		case "failed":
+			os.Exit(int(attempt.State.Failed.ExitCode))
+
+			return true
+
+		case "succeeded":
+			os.Exit(int(attempt.State.Succeeded.ExitCode))
+
+			return true
+		case "queued":
+			if looped {
+				fmt.Fprintf(out, "\033[1A\033[K")
+			}
+
+			fmt.Fprintf(out, "Queued on Job Runner Group %s\n", j.Spec.RunnerGroup)
+		case "running":
+			if looped {
+				fmt.Fprintf(out, "\033[1A\033[K")
+			}
+
+			lastCursor, err = logs.ShowLogs(ctx, cfg.API, out, jobName, cfg.Attach, lastCursor, 0)
```

Comment:
If there multiple `logs.ShowLogs()` errors in a row, this code is resetting the cursor.
We should only update `lastCursor` is `err == nil`.
e.g.:

```
Welcome 27 times (env TEST=)
Welcome 28 times (env TEST=)
Welcome 29 times (env TEST=)
Welcome 30 times (env TEST=)
Welcome 31 times (env TEST=)
Welcome 32 times (env TEST=)
Welcome 33 times (env TEST=)
Welcome 34 times (env TEST=)
Welcome 35 times (enHello this is the demo meeting
Welcome 1 times (env TEST=))                                 Error getting logs: 502 Bad Gateway: no active tunnel found for cluster
Welcome 2 times (env TEST=)no active tunnel found for clusterp://localhost:8080/api/v2/orgs/ddvcorp/jobs/basic-aqzp1ua": dial tcp 127.0.0.1:8080: connect: connection refused
Welcome 3 times (env TEST=)
Welcome 4 times (env TEST=)
Welcome 5 times (env TEST=)
Welcome 6 times (env TEST=)
Welcome 7 times (env TEST=)
Welcome 8 times (env TEST=)
Welcome 9 times (env TEST=)
Welcome 10 times (env TEST=)
```

Reply:
How can i reproduce this? I tried turning off my wifi, but i kept the connection open with the sse stream

Reply:
You can disconnect your cluster, and/or restart the `apiserver` (note that in my case, I'm using a local version).